pennylane/transforms/hamiltonian_expand.py

```diff
@@ -101,17 +101,21 @@ def hamiltonian_expand(tape, group=True):
         raise ValueError(
             "Passed tape must end in `qml.expval(H)`, where H is of type `qml.Hamiltonian`"
         )
+
     if group:
         hamiltonian.simplify()
         return qml.transforms.measurement_grouping(tape, hamiltonian.ops, hamiltonian.coeffs)
 
     # create tapes that measure the Pauli-words in the Hamiltonian
     tapes = []
     for ob in hamiltonian.ops:
-        new_tape = tape.copy()
-        new_tape._measurements = [
-            qml.measure.MeasurementProcess(return_type=qml.operation.Expectation, obs=ob)
-        ]
+        # we need to create a new tape here, because
+        # updating metadata of a copied tape is error-prone
+        # when the observables where changed
+        with tape.__class__() as new_tape:
+            for op in tape.operations:
+                qml.apply(op)
+            qml.expval(ob)
```

Comment:
I toggled between the deleted version and the new one before, but it turns out we need to create a new tape. Replacing measurements does not update the right meta info (even if I use `tape.update()`) and the circuit graph gets stuck!

Reply:
That's... concerning :fearful:

Reply:
A little :) I haven't dug deep though

Reply:
And in some sense this is intended behaviour, one should rather rerecord a tape than hack attributes...